gobblin-core/src/main/java/gobblin/http/ThrottledHttpClient.java

```diff
@@ -0,0 +1,66 @@
+package gobblin.http;
+
+import java.io.IOException;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+
+import com.typesafe.config.Config;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.broker.gobblin_scopes.GobblinScopeTypes;
+import gobblin.broker.iface.NotConfiguredException;
+import gobblin.broker.iface.SharedResourcesBroker;
+import gobblin.util.http.HttpLimiterKey;
+import gobblin.util.limiter.Limiter;
+import gobblin.util.limiter.broker.SharedLimiterFactory;
+
+
+/**
+ * A {@link HttpClient} for throttling calls to the underlying TX operation using the input
+ * {@link Limiter}.
+ */
+@Slf4j
+public abstract class ThrottledHttpClient<RQ, RP> implements HttpClient<RQ, RP>  {
+
+  @Getter
+  protected Limiter limiter = null;
+  @Setter
+  protected SharedResourcesBroker<GobblinScopeTypes> broker = null;
+  @Getter
+  protected Config config = null;
+
+  public ThrottledHttpClient (Config config, SharedResourcesBroker<GobblinScopeTypes> broker) {
+    this.config = config;
+    this.broker = broker;
+    createLimiter ();
+  }
+
+  public RP sendRequest(RQ request) throws IOException {
+    try {
+      if (limiter.acquirePermits(1) != null) {
+        log.info ("Acquired permits successfully");
+        return sendRequestImpl (request);
+      } else {
+        throw new RuntimeException ("Cannot reach here: acuirePermits should never return null");
```

Comment:
Acquire permits may actually return null if it determines this call can never be made.